### ISSUE TITLE
selector-no-qualifying-type reports error on line following the error

### ISSUE THREAD

**Issue by montchr** [stylelint/stylelint#2530]
> Describe the issue. Is it a bug or a feature request (new rule, new option, etc.)?

Errors reported by `selector-no-qualifying-type` are reported on the line following the actual error.

> Which rule, if any, is this issue related to?

`selector-no-qualifying-type`

> What CSS is needed to reproduce this issue?

```scss
      ul {
        &.last {
          /* ... */
        }
      }
```

But it's not just limited to `&`. The error for this is also reported on the following line:

```scss
html.last {
  /* ... */
}
```

> What stylelint configuration is needed to reproduce this issue?

```json
{
  "rules": {
    "selector-no-qualifying-type": true
  }
}
```

> Which version of stylelint are you using?

`7.10.1`

> How are you running stylelint: CLI, PostCSS plugin, Node API?

`grunt-stylelint` and the Atom plugin both report the same thing.

> Does your issue relate to non-standard syntax (e.g. SCSS, nesting, etc.)?

SCSS + nesting

> What did you expect to happen?

I expected `selector-no-qualifying-type` to report the error on the line it actually occurred on.

> What actually happened (e.g. what warnings or errors you are getting)?

The error is reported on the following line. Note that if the rule is on a single line the error is reported on the correct line:

```scss
&.last { /* ... */ }
```

---

A quick question I'd like to tack on at the end here: is there an equivalent to the [SCSSLint `ChainedClasses` linter](https://github.com/brigade/scss-lint/blob/master/lib/scss_lint/linter/README.md#chainedclasses) in Stylelint?

**Comment by montchr** [comment]
If I disable and re-enable the linter around the line causing the issue nothing is reported, as expected:

```scss
        // stylelint-disable selector-no-qualifying-type
        &.last {
          // stylelint-enable selector-no-qualifying-type
          border-right: 0;
        }
```

**Comment by evilebottnawi** [comment]
@montchr https://github.com/stylelint/stylelint/blob/master/CHANGELOG.md#head, just wait 8 release :smile:

**Comment by jeddy3** [comment]
@montchr Thanks for the report and for using the template.

I'm unable to replicate this, using either standard syntax or `scss`.

```css
html.last {
  /* ... */
}
```

Gives:

```
1:1  ✖  Unexpected qualifying type selector   selector-no-qualifying-type
```

And:

```css
ul {
  &.last {
    /* ... */
  }
}
```

Gives:

```
2:3  ✖  Unexpected qualifying type selector   selector-no-qualifying-type
```

The rule will report at the location _direct nesting selector_ when it's used.

Are you able to create a reproducible minimum test repo that illustrates the behaviour you're experiencing, please?

**Comment by jeddy3** [comment]
Closing. Feel free to reopen if you're able to create a reproducible test repo for us.

**Comment by dmikester1** [comment]
This is happening with me too.  Took me a while to figure out why it was yelling at me about code that looked fine.

**Comment by jeddy3** [comment]
@dmikester1 Can you please create a reproducible test repo so we can get to the bottom of this.
